include/wx/dc.h

```diff
@@ -1405,6 +1405,78 @@ class WXDLLIMPEXP_CORE wxDCTextColourChanger
     wxDECLARE_NO_COPY_CLASS(wxDCTextColourChanger);
 };
 
+// ----------------------------------------------------------------------------
+// helper class: you can use it to temporarily change the DC text background colour and
+// restore it automatically when the object goes out of scope
+// ----------------------------------------------------------------------------
+
+class WXDLLIMPEXP_CORE wxDCTextBgColourChanger
+{
+public:
+    wxDCTextBgColourChanger(wxDC& dc) : m_dc(dc), m_colBgOld() { }
+
+    wxDCTextBgColourChanger(wxDC& dc, const wxColour& col) : m_dc(dc)
+    {
+        Set(col);
+    }
+
+    ~wxDCTextBgColourChanger()
+    {
+        if ( m_colBgOld.IsOk() )
+            m_dc.SetTextBackground(m_colBgOld);
+    }
+
+    void Set(const wxColour& col)
+    {
+        if ( !m_colBgOld.IsOk() )
+            m_colBgOld = m_dc.GetTextBackground();
+        m_dc.SetTextBackground(col);
+    }
+
+private:
+    wxDC& m_dc;
+
+    wxColour m_colBgOld;
+
+    wxDECLARE_NO_COPY_CLASS(wxDCTextBgColourChanger);
+};
+
+// ----------------------------------------------------------------------------
+// helper class: you can use it to temporarily change the DC text background mode and
+// restore it automatically when the object goes out of scope
+// ----------------------------------------------------------------------------
+
+class WXDLLIMPEXP_CORE wxDCTextBgModeChanger
```

Comment:
Is the class name intuitive enough, the wrapped `wxDC` method does not have _Text_ in its name nor are any of the words abbreviated in the changer names or `wxDC` method names (this applies also to [wxDCTextBgColourChanger](url))? 

OTOH, according to the documentation, the mode is used only for drawing text and perhaps the full names would be too long for some...